### ISSUE TITLE
Switch to a different Bitcoin balance query api if blockchain.info fails

### ISSUE THREAD

**Issue by LefterisJP** [rotki/rotki#1828]
## Abstract

A user has reported that suddenly their bitcoin xpub balances are shown as 0. After inspection of the logs it seems that the blockchain.info call failed with invalid json.

Logs:

```
ERROR rotkehlchen.rotkehlchen: Querying blockchain balances failed due to: bitcoin external API request for balances failed due to https://blockchain.info/multiaddr?active=XXXXX|YYYYY|ZZZZZ....|FFFFF returned malformed json. Error: Expecting value: line 1 column 1 (char 0)
```

The user queried 170 addresses in 1 call. Which should not be a problem under normal circumstances if I recall. Also the docs don't mention it as a problem.

Trying it from the browswer you can a 502 error as of the writing of this issue.

![2020-11-27_23-38](https://user-images.githubusercontent.com/1658405/100487674-2d0a0f00-310a-11eb-9192-d76f865c611b.png)

## Task

Offer an alternative call via the blockstream api if the blockchain.info api call fails.

**Comment by kilrau** [comment]
Is this still relevant or rather vice versa? In my rotki debug logs I can only see calls to blockstream.info.

**Comment by LefterisJP** [comment]
@kilrau yes I think it still is. But perhaps should be renamed to generally try many sources when one fails. The reason you only see blockstream.info is that blockchain.info does not accept segwit bc1 addresses and I assume all your addresses, being a modern BTC user, are of that type.

**Comment by kilrau** [comment]
True that. You really don't want to use blockchain.info as anything these days ;)

Mempool.space is another great service, never used their API but looks like the address endpoint has everything you'll need: https://mempool.space/api
